### PR TITLE
Revert: remove feature the out of stock label in order to prepare the availability feature on FO

### LINKED ISSUE #24759: Remove the “out of stock flag on product listings and product pages + the toggle in BO product settings page
#### Is your feature request related to a problem?

As a visitor, I want to be able to see easily which product are available for order when I am in a product list

#### Describe the solution you'd like

Remove the “out of stock flag on product listings and product pages + the toggle "Display out-of-stock label on product listing pages" in BO > shop parameters > product settings page that was introduced in this [PR](https://github.com/PrestaShop/PrestaShop/pull/21751)

The out of stock label will be displayed in another way, see details in these issues:
https://github.com/PrestaShop/PrestaShop/issues/24758
https://github.com/PrestaShop/PrestaShop/issues/23659

### PR DESCRIPTION
**Pull request by NeOMakinG**
This reverts commit d856042149a7ed81587c6e2002a3165e7e936a82.

| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | We need to revert the out of stock label stuff, because we're preparing an availability feature on FO
| Type?             | refacto
| Category?         | FO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #24759.
| How to test?      | See https://github.com/PrestaShop/PrestaShop/pull/21751, it shouldn't be working anymore
| Possible impacts? | No more out of stock labels on FO

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/25050)

### CODE REVIEW

File: install-dev/upgrade/sql/1.7.8.0.sql [hunk 1]
```diff
@@ -35,7 +35,6 @@ ALTER TABLE `PREFIX_employee` ADD `has_enabled_gravatar` TINYINT UNSIGNED DEFAUL
 
 INSERT INTO `PREFIX_configuration` (`name`, `value`, `date_add`, `date_upd`) VALUES
     ('PS_COOKIE_SAMESITE', 'Lax', NOW(), NOW()),
-    ('PS_SHOW_LABEL_OOS_LISTING_PAGES', '1', NOW(), NOW()),
```
Comment (Progi1984) [inline install-dev/upgrade/sql/1.7.8.0.sql:38]: Please revert this line

Comment (Progi1984) [reply install-dev/upgrade/sql/1.7.8.0.sql:38]: Add a delete in 1790.sql

File: src/Adapter/Presenter/Cart/CartPresenter.php [hunk 2]
```diff
@@ -111,6 +111,15 @@ private function presentProduct(array $rawProduct)
...
 
+        $settings = new ProductPresentationSettings();
+
+        $settings->catalog_mode = Configuration::isCatalogMode();
+        $settings->catalog_mode_with_prices = (int) Configuration::get('PS_CATALOG_MODE_WITH_PRICES');
+        $settings->include_taxes = $this->includeTaxes();
+        $settings->allow_add_variant_to_cart_from_listing = (int) Configuration::get('PS_ATTRIBUTE_CATEGORY_DISPLAY');
+        $settings->stock_management_enabled = Configuration::get('PS_STOCK_MANAGEMENT');
+        $settings->showPrices = Configuration::showPrices();
+
```
Comment (Progi1984) [inline src/Adapter/Presenter/Cart/CartPresenter.php]:
```suggestion
```

File: src/Adapter/Presenter/Product/ProductLazyArray.php [hunk 3]
```diff
@@ -104,9 +93,7 @@ public function __construct(
         Link $link,
         PriceFormatter $priceFormatter,
         ProductColorsRetriever $productColorsRetriever,
-        TranslatorInterface $translator,
-        HookManager $hookManager = null,
-        Configuration $configuration = null
```
Comment (Progi1984) [inline src/Adapter/Presenter/Product/ProductLazyArray.php:109]: BC Break, please document it

Comment (Progi1984) [inline src/Adapter/Presenter/Product/ProductLazyArray.php:109]: Please  don't remove it

File: src/Adapter/Presenter/Product/ProductLazyArray.php [hunk 4]
```diff
@@ -605,59 +580,18 @@ private function shouldShowPrice(
...
     {
         return (bool) $product['available_for_order'];
     }
 
-    /**
-     * @param array $product
-     *
-     * @return bool
-     */
-    private function shouldShowOutOfStockLabel(ProductPresentationSettings $settings, array $product): bool
```
Comment (Progi1984) [inline src/Adapter/Presenter/Product/ProductLazyArray.php:620]: In this file only remove this method (and its call)

File: src/Adapter/Presenter/Product/ProductPresenter.php [hunk 5]
```diff
@@ -29,8 +29,6 @@
 use Hook;
 use Language;
 use Link;
-use PrestaShop\PrestaShop\Adapter\Configuration;
```
Comment (Progi1984) [inline src/Adapter/Presenter/Product/ProductPresenter.php:32]: Please don't revert all these changes

File: src/Core/Product/ProductPresentationSettings.php [hunk 6]
```diff
@@ -36,10 +36,6 @@ class ProductPresentationSettings
     public $stock_management_enabled;
     public $showPrices;
     public $lastRemainingItems;
-    /**
-     * @var bool|null
-     */
-    public $showLabelOOSListingPages;
```
Comment (Progi1984) [inline src/Core/Product/ProductPresentationSettings.php:42]: BC Break : remove public attribute

**Review by Progi1984 — COMMENTED** [review]
Some feedbacks

**Comment by PierreRambaud** [comment]
I'm closing it as work has to be done when the new behavior will be added.

**Pull request closed without merging**